### PR TITLE
Feature/283 update rn

### LINKED ISSUE #283: Update to RN 0.65

### PR DESCRIPTION
**Pull request by ahoseinian**
it's building now on my local machine on both platforms. I'll test it again after bitrise build to see everything is working.

### CODE REVIEW

File: ios/Litentry.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -531,12 +531,12 @@
...
...$_JSON_OUTPUT_BASE64\")\n  echo \"warning:   A firebase.json file was not found, whilst this file is optional it is recommended to include it to configure firebase services in React Native Firebase.\"\nfi;\n\necho \"info: 2) Injecting Info.plist entries: \"\n\n# Log out the keys we're adding\nfor i in \"${!_PLIST_ENTRY_KEYS[@]}\"; do\n  echo \"    ->  $i) ${_PLIST_ENTRY_KEYS[$i]}\" \"${_PLIST_ENTRY_TYPES[$i]}\" \"${_PLIST_ENTRY_VALUES[$i]}\"\ndone\n\nfor plist in \"${_TARGET_PLIST}\" \"${_DSYM_PLIST}\" ; do\n  if [[ -f \"${plist}\" ]]; then\n\n    # paths with spaces break the call to setPlistValue. temporarily modify\n    # the shell internal field separator variable (IFS), which normally\n    # includes spaces, to consist only of line breaks\n    oldifs=$IFS\n    IFS=\"\n\"\n\n    for i in \"${!_PLIST_ENTRY_KEYS[@]}\"; do\n      setPlistValue \"${_PLIST_ENTRY_KEYS[$i]}\" \"${_PLIST_ENTRY_TYPES[$i]}\" \"${_PLIST_ENTRY_VALUES[$i]}\" \"${plist}\"\n    done\n\n    # restore the original internal field separator value\n    IFS=$oldifs\n  else\n    echo \"warning:   A Info.plist build output file was not found (${plist})\"\n  fi\ndone\n\necho \"info: <- RNFB build script finished\"\n";
```
Comment (silva-fj) [inline ios/Litentry.xcodeproj/project.pbxproj:539]: Was this changed by xcode?

File: package.json [hunk 2]
```diff
@@ -44,8 +44,8 @@
     "process": "^0.11.10",
     "proxy-polyfill": "^0.3.2",
     "qrcode-generator": "^1.4.4",
-    "react": "17.0.1",
-    "react-native": "0.64.2",
+    "react": "17.0.2",
+    "react-native": "0.65.0",
```
Comment (silva-fj) [inline package.json]: Did you try upgrading to the [latest version](https://github.com/facebook/react-native/releases/tag/v0.65.1)?

File: src/screen/tips/TipDetailScreen.tsx [hunk 3]
```diff
@@ -128,7 +128,7 @@ function TipDetailScreen({route}: ScreenProps) {
         data={tip.tips}
         style={[globalStyles.paddedContainer, styles.container]}
         ItemSeparatorComponent={Divider}
-        renderItem={({item}) => {
+        renderItem={({item}: {item: any[]}) => {
```
Comment (silva-fj) [inline src/screen/tips/TipDetailScreen.tsx]: Why are we having this in every PR? 😆

Comment (ahoseinian) [reply src/screen/tips/TipDetailScreen.tsx]: it doesn't let me commit if I don't fix this :D

File: package.json (hunk 2 above)
Comment (silva-fj) [reply package.json]: Can you also update `@types/react-native` and `@types/react` to the latest versions?

File: ios/Litentry.xcodeproj/project.pbxproj (hunk 1 above)
Comment (ahoseinian) [reply ios/Litentry.xcodeproj/project.pbxproj:539]: yes, as far as I remember I didn't change this manually

File: src/screen/tips/TipDetailScreen.tsx (hunk 3 above)
Comment (ahoseinian) [reply src/screen/tips/TipDetailScreen.tsx]: done ✅

File: package.json (hunk 2 above)
Comment (ahoseinian) [reply package.json]: done ✅

**Review by silva-fj — DISMISSED** [review]

**Review by silva-fj — APPROVED** [review]

**Pull request merged**